### PR TITLE
Use headless notebooks for training facebook-opt-125m

### PR DESCRIPTION
**Pull request by nstogner**
This gives us a nice way to view the output of a training job and visualize loss graphs, etc (that the notebook can produce while training).

### CODE REVIEW

**Comment by nstogner** [comment]
NOTE: Github can render `.ipynb` files: https://github.com/substratusai/models/blob/headless-notebook-training/facebook-opt-125m/train.ipynb

**Review by samos123 — APPROVED** [review]

File: facebook-opt-125m/scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+
+set -xe
+
+jupyter nbconvert --debug --to notebook --execute /model/src/build.ipynb --output /model/logs/build.ipynb
```
Comment (samos123) [inline facebook-opt-125m/scripts/build.sh:5]: shoudln't this be --to python instead of --to notebook?

Comment (samos123) [reply facebook-opt-125m/scripts/build.sh:5]: ignore I see this is a way of executing the notebook: https://nbconvert.readthedocs.io/en/latest/usage.html#convert-notebook

File: facebook-opt-125m/README.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ Run Server.
 
 ```sh
 # Run longrunning HTTP API.
-docker run -p 8080:8080 facebook-opt-125m python serve.py
+docker run -it -p 8080:8080 facebook-opt-125m ./serve.sh
```
Comment (samos123) [inline facebook-opt-125m/README.md:33]: isn't  serve.sh under /model/scripts in your case? so shouldn't it be `./scripts/serve.sh` or just plain `serve.sh`

File: facebook-opt-125m/README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ Explore with a Notebook.
 
 ```sh
 # Run Jupyter Notebook server.
-docker run -p 8888:8888 facebook-opt-125m jupyter notebook --allow-root --ip=0.0.0.0 --NotebookApp.token='' --notebook-dir='/app'
+docker run -it -p 8888:8888 facebook-opt-125m ./dev.sh
```
Comment (samos123) [inline facebook-opt-125m/README.md:23]: I think this path is incorrect

**Review by samos123 — CHANGES_REQUESTED** [review]

**Comment by samos123** [comment]
curious, how do you get the ipynb files out from the container after running dev.sh? The use case would be to work on the train.ipynb, get it to work, download train.ipynb from container to local directory, git add train.ipynb.

My naive way would be to call `docker cp` to get the `train.ipynb` file out from container into local dir.

File: facebook-opt-125m/README.md (hunk 3 above)
Comment (samos123) [reply facebook-opt-125m/README.md:23]: I just tested this and seems it's incorrect at least based on me trying out falcon-7b-instruct. It should be `dev.sh` or `./scripts/dev.sh`

File: facebook-opt-125m/README.md (hunk 2 above)
Comment (nstogner) [reply facebook-opt-125m/README.md:33]: Good catch, it should be ‘serve.sh’

File: facebook-opt-125m/README.md (hunk 3 above)
Comment (nstogner) [reply facebook-opt-125m/README.md]: Agreed, solved

**Comment by nstogner** [comment]
Updated the README. You can now develop with the notebook b/c the docker container mounts `./src` in the README steps.

**Review by samos123 — APPROVED** [review]

**Pull request merged**